lib/spec/spec.go

```diff
@@ -99,8 +102,17 @@ func NewVirtualMachineConfigSpec(ctx context.Context, session *session.Session,
 	log.Debugf("Adding metadata to the configspec: %+v", config.Metadata)
 	// TEMPORARY
 
+	// set VM name to prettyname-ID, to make it readable a little bit
+	// if prettyname-ID is longer than max vm name length, truncate pretty name, instead of UUID, to make it unique
+	nameMaxLen := MaxVMNameLength - len(config.ID)
+	prettyName := config.Name
+	if len(prettyName) > nameMaxLen-1 {
+		prettyName = prettyName[:nameMaxLen-1]
+	}
+	fullName := fmt.Sprintf("%s-%s", prettyName, config.ID)
 	s := &types.VirtualMachineConfigSpec{
-		Name: config.ID,
+		Name:         fullName,
+		InstanceUuid: config.InstanceUUID,
```

Comment:
@dougm Do you mean set Uuid, instead of InstanceUuid?
But the search is based on instance uuid, instead of uuid, right? How can that work?